src/WooCommerce.php

```diff
@@ -1509,8 +1511,56 @@ private static function getRepricingOptionsDropdownValues(\WC_Product $product)
       !empty($brand) ? $brand . ' | no repricing' : 'no repricing' => __('No Repricing', Plugin::L10N),
       !empty($brand) ? $brand . ' | lower prices only' : 'lower prices only' => __('Lower prices only', Plugin::L10N),
     ];
-    
+
     return $dropdown_values;
   }
 
+  /**
+   * Calculates shipping price + tax
+   *
+   * @param  object $shipping_method
+   * @return float
+   */
+  public static function getTotalShippingPrice(object $shipping_method):float {
+    $shipping_price = floatval($shipping_method->get_total());
```

Comment:
Some shipping methods don't have a price, they are free, so let's add an if statement and return early if that is the case.

Reply:
```suggestion
    $shipping_price = floatval($shipping_method->get_total());
    if (!$shipping_price) {
     return "";
    }
```